// ng2-material/src/core/util/viewport.js
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var core_1 = require("@angular/core");
/**
 * Interact with the window and document in a cross-platform friendly way.
 *
 * Rather than interacting with the window, inject this service and use it.
 */
var ViewportHelper = (function () {
    function ViewportHelper() {
    }
    return ViewportHelper;
}());
ViewportHelper = __decorate([
    core_1.Injectable()
], ViewportHelper);
exports.ViewportHelper = ViewportHelper;
var BrowserViewportHelper = (function (_super) {
    __extends(BrowserViewportHelper, _super);
    function BrowserViewportHelper() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    BrowserViewportHelper.prototype.getDocumentNativeElement = function () {
        return window.document;
    };
    BrowserViewportHelper.prototype.requestFrame = function (fn) {
        return window.requestAnimationFrame(fn);
    };
    BrowserViewportHelper.prototype.matchMedia = function (query) {
        return window.matchMedia(query);
    };
    BrowserViewportHelper.prototype.scrollTop = function () {
        return window.pageYOffset || document.documentElement.scrollTop;
    };
    return BrowserViewportHelper;
}(ViewportHelper));
exports.BrowserViewportHelper = BrowserViewportHelper;
var NodeViewportMediaMatch = (function () {
    function NodeViewportMediaMatch(matches, media) {
        if (matches === void 0) { matches = false; }
        if (media === void 0) { media = ''; }
        this.matches = matches;
        this.media = media;
    }
    NodeViewportMediaMatch.prototype.addListener = function (listener) {
    };
    NodeViewportMediaMatch.prototype.removeListener = function (listener) {
    };
    return NodeViewportMediaMatch;
}());
exports.NodeViewportMediaMatch = NodeViewportMediaMatch;
var NodeViewportHelper = (function (_super) {
    __extends(NodeViewportHelper, _super);
    function NodeViewportHelper() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    NodeViewportHelper.prototype.getDocumentNativeElement = function () {
        return {};
    };
    NodeViewportHelper.prototype.requestFrame = function (fn) {
        return process.nextTick(fn);
    };
    NodeViewportHelper.prototype.matchMedia = function (query) {
        return new NodeViewportMediaMatch(false, query);
    };
    NodeViewportHelper.prototype.scrollTop = function () {
        return 0;
    };
    return NodeViewportHelper;
}(ViewportHelper));
exports.NodeViewportHelper = NodeViewportHelper;
