// squeezebox/tests/index.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var AppComponent = (function () {
    function AppComponent() {
        this.itemsList = [
            {
                title: 'Item 1',
                description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident'
            },
            {
                title: 'Item 2',
                description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident'
            }
        ];
    }
    return AppComponent;
}());
AppComponent = __decorate([
    core_1.Component({
        selector: 'app',
        template: "\n        <div>\n            <squeezebox [multiple]=\"false\">\n                <sb-item  *ngFor=\"let item of itemsList\">\n                    <sb-item-head>{{item.title}}</sb-item-head>\n                    <sb-item-body>{{item.description}}</sb-item-body>\n                </sb-item>\n            </squeezebox>\n        </div>\n\n        <h3>Already expanded</h3>\n        <div>\n            <squeezebox [multiple]=\"false\">\n                <sb-item>\n                    <sb-item-head>Item 1</sb-item-head>\n                    <sb-item-body>Lorem ipsum dolor</sb-item-body>\n                </sb-item>\n                <sb-item [collapsed]=\"false\">\n                    <sb-item-head>Item 2</sb-item-head>\n                    <sb-item-body>Lorem ipsum dolor</sb-item-body>\n                </sb-item>\n            </squeezebox>\n        </div>\n    "
    }),
    __metadata("design:paramtypes", [])
], AppComponent);
exports.AppComponent = AppComponent;
