// ng2-material/src/index.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var core_1 = require("@angular/core");
var content_1 = require("./components/content/content");
var index_1 = require("./components/data-table/index");
var divider_1 = require("./components/divider/divider");
var index_2 = require("./components/pagination/index");
var peekaboo_1 = require("./components/peekaboo/peekaboo");
var subheader_1 = require("./components/subheader/subheader");
var media_1 = require("./core/util/media");
var viewport_1 = require("./core/util/viewport");
var dialog_module_1 = require("./components/dialog/dialog.module");
var backdrop_1 = require("./components/backdrop/backdrop");
var util_module_1 = require("./core/util/util.module");
// import {MdMessage, MdMessages} from "./components/form/messages"; // Needs upgrade from deprecated forms module
__export(require("./components/backdrop/backdrop"));
__export(require("./components/content/content"));
__export(require("./components/data-table/index"));
__export(require("./components/dialog/index"));
__export(require("./components/divider/divider"));
__export(require("./components/pagination/index"));
__export(require("./components/peekaboo/peekaboo"));
__export(require("./components/subheader/subheader"));
__export(require("./core/util/media"));
__export(require("./core/util/viewport"));
__export(require("./core/util/animate"));
/**
 * Material Design component providers for use in a Node.JS environment.
 */
exports.MATERIAL_NODE_PROVIDERS = [
    { provide: viewport_1.ViewportHelper, useClass: viewport_1.NodeViewportHelper },
    media_1.Media
];
/**
 * Material Design component providers for use in the browser.
 */
exports.MATERIAL_BROWSER_PROVIDERS = exports.MATERIAL_NODE_PROVIDERS.concat([
    { provide: viewport_1.ViewportHelper, useClass: viewport_1.BrowserViewportHelper },
]);
/**
 * Collection of Material Design component modules.
 */
var MATERIAL_MODULES = [
    backdrop_1.MdBackdropModule,
    content_1.MdContentModule,
    index_1.MdDataTableModule,
    dialog_module_1.MdDialogModule,
    divider_1.MdDividerModule,
    index_2.MdPaginationModule,
    peekaboo_1.MdPeekabooModule,
    subheader_1.MdSubheaderModule,
    util_module_1.MdServicesModule
];
/**
 * Material Design module for use in a Node.JS environment.
 */
var Ng2MaterialNodeModule = Ng2MaterialNodeModule_1 = (function () {
    function Ng2MaterialNodeModule() {
    }
    Ng2MaterialNodeModule.forRoot = function () {
        return {
            ngModule: Ng2MaterialNodeModule_1,
            providers: exports.MATERIAL_NODE_PROVIDERS
        };
    };
    return Ng2MaterialNodeModule;
}());
Ng2MaterialNodeModule = Ng2MaterialNodeModule_1 = __decorate([
    core_1.NgModule({
        exports: MATERIAL_MODULES,
        imports: MATERIAL_MODULES
    })
], Ng2MaterialNodeModule);
exports.Ng2MaterialNodeModule = Ng2MaterialNodeModule;
/**
 * Material Design module for use in the browser.
 */
var Ng2MaterialModule = Ng2MaterialModule_1 = (function () {
    function Ng2MaterialModule() {
    }
    Ng2MaterialModule.forRoot = function () {
        return {
            ngModule: Ng2MaterialModule_1,
            providers: exports.MATERIAL_BROWSER_PROVIDERS
        };
    };
    return Ng2MaterialModule;
}());
Ng2MaterialModule = Ng2MaterialModule_1 = __decorate([
    core_1.NgModule({
        exports: MATERIAL_MODULES,
        imports: MATERIAL_MODULES
    })
], Ng2MaterialModule);
exports.Ng2MaterialModule = Ng2MaterialModule;
var Ng2MaterialNodeModule_1, Ng2MaterialModule_1;
